tiatoolbox/models/architecture/sccnn.py

```diff
@@ -0,0 +1,360 @@
+        model is retrained in torch as the original model was trained in Matlab. This
+        model also relies on RGB image as input. The original model uses HRGB as input,
+        where 'H' represents hematoxylin.
+
+    The tiatoolbox model should produce the following results on the CRC dataset using
+    10 pixels as radius for true detection:
+
+    .. list-table:: MicroNet performance
+       :widths: 15 15 15 15 15
+       :header-rows: 1
+
+       * - Model name
+         - Data set
+         - Precision
+         - Recall
+         - F1Score
+       * - sccnn-crchisto
+         - CRCHisto
+         - 0.79
+         - 0.81
+         - 0.80
+
+    Args:
+        num_input_channels (int):
+            Number of channels in input. default=3.
+        out_height (int):
+            Output height. default=13.
+        out_width (int):
+            Output width. default=13.
+        radius (int):
+            Radius for nucleus detection, default = 12.
+
+    References:
+        [1] Sirinukunwattana, Korsuk, et al.
+        "Locality sensitive deep learning for detection and classification
+        of nuclei in routine colon cancer histology images."
+        IEEE transactions on medical imaging 35.5 (2016): 1196-1206.
+
+    """
+
+    def __init__(
+        self,
+        num_input_channels: int = 3,
+        out_height: int = 13,
+        out_width: int = 13,
+        radius: int = 12,
+    ):
+        super().__init__()
+        self.in_ch = num_input_channels
+        self.out_height = out_height
+        self.out_width = out_width
+
+        # Create mesh grid and convert to 3D vector
+        x, y = torch.meshgrid(
+            torch.range(0, out_height - 1), torch.range(0, out_width - 1)
+        )
+        self.x = torch.unsqueeze(x, dim=0).type(torch.float32)
+        self.y = torch.unsqueeze(y, dim=0).type(torch.float32)
+
+        self.radius = radius
+
+        def conv_act_branch(in_ch: int, out_ch: int, k_dim: int) -> torch.nn.ModuleDict:
+            """Convolution and Activation branch for SCCNN.
+
+            Args:
+                in_ch (int):
+                    Number of channels in input.
+                out_ch (int):
+                    Number of required channels in output.
+                k_dim (int):
+                    Kernel size of convolution filter.
+
+            Returns:
+                torch.nn.ModuleDict:
+                    Module dictionary.
+
+            """
+            module_dict = OrderedDict()
+            module_dict["conv1"] = nn.Sequential(
+                nn.Conv2d(
+                    in_ch,
+                    out_ch,
+                    kernel_size=(k_dim, k_dim),
+                    stride=(1, 1),
+                    padding=0,
+                    bias=True,
+                ),
+                nn.ReLU(),
+            )
+
+            return nn.ModuleDict(module_dict)
+
+        def sc(in_ch, out_ch) -> torch.nn.ModuleDict:
```

Comment:
Names could be better here. Would suggest just using more verbose but clearer names e.g. `spatially_constrained_layer`, `in_channels` etc.

Reply:
Fixed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/434/commits/6dcf8731a6dc27caa2cafd10ed24502253ecb530

Reply:
Fixed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/434/commits/0b7bd23409f8299b3fe723d73a1131b518fe6b97